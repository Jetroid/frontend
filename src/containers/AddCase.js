import CaseForm from '../components/CaseForm'
import { reduxForm } from 'redux-form'

const mapStateToProps = (state /* , ownProps */) => {
  return state.form
}

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    onSubmit: (data) => {
      console.log('FORM PROPS', data, ownProps)
    }
  }
}

export const fields = [ 'firstName', 'lastName', 'age', 'color', 'bio' ]

let AddCaseForm = reduxForm({
  form: 'ncase',
  fields: ['firstName', 'lastName', 'email']
}, mapStateToProps, mapDispatchToProps)(CaseForm)

export default AddCaseForm
