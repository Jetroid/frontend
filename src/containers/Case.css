.main-contents {
  min-height: 200px;
}

.editButton {
  position: fixed;
  bottom: 2em;
  right: 2em;
}



.detailed-case-component {
  margin: 0 auto;
  overflow: hidden;
  padding-bottom: 32px;
  padding-top: 64px;
}

.detailed-case-component .sidebar {
  padding: 0 32px;
  width: 200px;
  float: left;
  margin-top: 48px;
}

.detailed-case-component .sidebar .case-map {
  width: 100%;
}

.detailed-case-component .sidebar .case-location {
  font-weight: 500;
  color: #3f51b2;
  margin-top: 16px;
  margin-bottom: 16px;
}

.detailed-case-component .sidebar .progress-bar {
  width: 100%;
  background-color: #ddd;
  height: 16px;
}

.detailed-case-component .sidebar .progress-bar .progress-fill {
  height: 16px;
  background-color: #3f51b2;
}

.detailed-case-component .sidebar .progress-complete {
  font-size: 14px;
  margin-top: 8px;
  font-weight: 400;
}

.detailed-case-component .sidebar .sub-heading {
  font-size: 32px;
  font-weight: 400;
  line-height: 40px;
  margin-bottom: 12px;
  margin-top: 16px;
}

.detailed-case-component .sidebar .sub-sub-heading {
  font-size: 14px;
  font-weight: 500;
  color: #777;
  line-height: 18px;
  margin-bottom: 8px;
  margin-top: 16px;
}

.detailed-case-component .sidebar .tags a {
  display: block;
  font-size: 14px;
  color: #888;
  font-weight: 400;
  line-height: 18px;
  margin-bottom: 4px;
}

.detailed-case-component .sidebar .related-content a {
  font-size: 14px;
  font-weight: 500;
  color: #777;
  line-height: 18px;
  margin-bottom: 8px;
  margin-top: 8px;
  position: relative;
  display: block;
  padding-left: 32px;
}

.detailed-case-component .sidebar .related-content a:before {
  content: "";
  background-image: url("/img/arrow-down-icon.png");
  display: block;
  position: absolute;
  left: 0;
  width: 24px;
  height: 24px;
  opacity: 0.5;
  margin-top: -4px;
}

.detailed-case-component .main-area {
  /*float: right;*/
  width: 1248px;
  padding-right: 32px;
}

.detailed-case-component .main-area .case-box {
  width: 960px;
  float: left;
  padding: 32px 0;
}

.detailed-case-component .main-area .case-box .category {
  color: #ff6f00;
  font-size: 48px;
  line-height: 60px;
  font-weight: 500;
  margin-bottom: 8px;
}

.detailed-case-component .main-area .case-box .case-title {
  color: #3f51b2;
  font-size: 48px;
  line-height: 60px;
  font-weight: 500;
  margin-bottom: 48px;
  width: 780px;
}

.detailed-case-component .main-area .case-box .case-images {
  overflow: hidden;
}

.detailed-case-component .main-area .case-box .case-images .thumbnail {
  width: 288px;
  height: 160px;
  background-size: cover;
  background-position: center;
  margin-bottom: 10px;
  background-color: #eee;
  float: left;
  margin-right: 32px;
}

.detailed-case-component .main-area .case-box .authorship-details {
  margin-top: 32px;
  margin-bottom: 32px;
}

.detailed-case-component .main-area .case-box .authorship-details .author-line {
  font-size: 14px;
  font-weight: 500;
  color: #777;
  line-height: 18px;
}

.detailed-case-component .main-area .case-box .authorship-details .author-line a {
  color: #ff6f00;
  font-weight: 500;
}

.detailed-case-component .main-area .case-box .authorship-details .date-line {
  font-size: 14px;
  color: #777;
  font-weight: 400;
  line-height: 18px;
  margin-bottom: 6px;
}

.detailed-case-component .main-area .case-box .case-html h3 {
  font-size: 32px;
  font-weight: 400;
  line-height: 40px;
  margin-bottom: 12px;
  margin-top: 32px;
}

.detailed-case-component .main-area .case-box .case-html p,
.detailed-case-component .main-area .case-box .case-html li {
  font-size: 16px; /* XXX */
  font-weight: 100;
  line-height: 24px;
  margin-bottom: 16px;
}

.detailed-case-component .main-area .case-box .case-html h2 {
  font-size: 20px;
  font-weight: 400;
  line-height: 24px;
  margin-bottom: 16px;
}

.detailed-case-component .main-area .case-box .case-html li {
  list-style-type: square;
  margin-left: 32px;
  margin-bottom: 8px;
  padding-left: 8px;
}

.detailed-case-component .main-area .case-box .case-html ol li {
  list-style-type: lower-roman;
}

.detailed-case-component .main-area .case-box .case-html a {
  color: #ff6f00;
  font-weight: 400;
}

.detailed-case-component .main-area .case-tools {
  position: fixed;
  right: 6em;
  top: 4em;
  z-index: 12000;
  /*float: right;
  margin-right: 32px;
  padding-right: 32px;*/
}

.detailed-case-component .main-area .case-tools .top-icons {
  margin-top: 40px;
  position: fixed;
}

.detailed-case-component .main-area .case-tools .top-icons a {
  display: block;
  margin-bottom: 16px;
}

.detailed-case-component .main-area .case-tools .top-icons a img {
  width: 40px;
}

.detailed-case-component .main-area .case-tools .top-icons a:first-child img {
  width: 32px;
  margin: 4px;
}

.detailed-case-component .main-area .case-tools .bottom-icons {
  position: absolute;
  bottom: 0;
  margin-bottom: 48px;
  position: fixed;
}

.detailed-case-component .main-area .case-tools .bottom-icons a {
  position: relative;
  display: block;
}

.detailed-case-component .main-area .case-tools .bottom-icons a img {
  width: 80px;
  margin-left: -20px;
}

.detailed-case-component .main-area .case-tools .bottom-icons a::before {
  content: "";
  background-color: #ff6f00;
  position: absolute;
  color: #fff;
  font-weight: 400;
  left: 65px;
  top: 19px;
  width: 12px;
  height: 12px;
  transform: rotate(45deg);
  display: none;
}

.detailed-case-component .main-area .case-tools .bottom-icons a::after {
  content: "Edit";
  background-color: #ff6f00;
  padding: 6px 12px;
  position: absolute;
  display: none;
  color: #fff;
  font-weight: 400;
  left: 70px;
  top: 10px;
}

.detailed-case-component .main-area .case-tools .bottom-icons a:hover::before,
.detailed-case-component .main-area .case-tools .bottom-icons a:hover::after {
  display: block;
}
