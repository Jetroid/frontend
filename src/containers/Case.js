import React from "react";
import { bool, object } from "prop-types";
import { connect } from "react-redux";
import { injectIntl, intlShape } from "react-intl";
import api from "../utils/api";
import CaseDetails from "../components/CaseDetails";
import ItemFetcher from "./ItemFetcher";

function mapStateToProps({ auth }) {
  const { isAuthenticated } = auth;

  return {
    isAuthenticated
  };
}

export class Case extends React.Component {
  toggleFeatured(thing, featured) {
    console.log("setting featured for ", thing, " to ", featured);
    // XXX needs API to exist.
  }
  render() {
    return (
      <ItemFetcher
        api={api.fetchCaseById}
        id={Number(this.props.match.params.nodeID)}
        details={CaseDetails}
        toggleFeatured={this.toggleFeatured.bind(this)}
        {...this.props}
      />
    );
  }
}
Case.propTypes = {
  intl: intlShape.isRequired,
  isAuthenticated: bool,
  location: object
};

export default connect(mapStateToProps)(injectIntl(Case));
