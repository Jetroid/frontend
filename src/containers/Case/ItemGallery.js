import React from 'react';
import ImageGallery from 'react-image-gallery';

class ItemGallery extends React.Component {

  constructor(props) {
    super(props);
    this.defineImage = this.defineImage.bind(this);
    this.getWidth = this.getWidth.bind(this);
    this.renderItem = this.renderItem.bind(this);
  }

  componentDidMount() {
    this.props.loadGalleryPhotos();
  }


  defineImage(url){
    if (url){
      let width = this.getWidth(url);
      if (width < 500) {
        return ('portrait');
      } else {
        return ('landscape');
      }
    } else {
      return;
    }
  }

  getWidth(url){
    var img = new Image();
    img.src = url;
    return img.naturalWidth
  }

  renderItem(item) {
    const imageClass = this.defineImage(item.original);
    return (
      <div className='image-gallery-image'>
        <img
          className={imageClass}
          src={item.original}
          alt={item.originalAlt}
          srcSet={item.srcSet}
          sizes={item.sizes}
        />
      </div>
    );
  }

  render() {

    const images = this.props.items.map((photo) => {
      const src = photo;
      return ({
        original: src,
        thumbnail: src,
      });
    });

    return (
      <ImageGallery
        items={images}
        showFullscreenButton={false}
        renderItem={this.renderItem}
        slideInterval={2000} />
    );
  }

}

export default ItemGallery;
