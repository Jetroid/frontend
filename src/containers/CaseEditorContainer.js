import React, { Component } from "react";
import CaseEditor from "../components/CaseEditor/CaseEditor";
import api from "../utils/api";

// May make sense to revert back to redux for this work. this is duplicative.
// Merge with EditCase, which is similar but not currently being used.

export default class CaseEditorContainer extends Component {
  getNouns() {
    let component = this;
    let nouns = {
      CASE: "cases",
      ORGANIZATION: "organizations",
      METHOD: "methods"
    };
    Object.keys(nouns).map((k, i) =>
      api.fetchNouns(k).then(function(the_nouns) {
        component.setState({ [nouns[k]]: the_nouns });
      }));
  }

  componentWillMount() {
    let component = this;
    api.fetchCaseById(this.props.params.nodeID).then(function(thing) {
      if (thing) {
        thing.start_date = new Date(thing.start_date);
        thing.end_date = new Date(thing.end_date);
        thing.updated_date = new Date(thing.updated_date);
      }
      component.setState({ case: thing });
    });
    this.getNouns();
  }

  onSubmit(values, state, props, instance) {
    let router = this.props.router;
    let location = this.props.location;
    api.saveCase(state).then(function(thing) {
      let pathparts = location.pathname.split("/");
      pathparts.pop();
      router.push(pathparts.join("/"));
    });
  }

  render() {
    if (
      this.state &&
      this.state.case &&
      this.state.cases &&
      this.state.methods &&
      this.state.organizations
    ) {
      // need to turn this into a loop
      let casesArr = Object.keys(this.state.cases).map(k => k);
      let methodsArr = Object.keys(this.state.methods).map(k => k);
      let orgsArr = Object.keys(this.state.organizations).map(k => k);
      return (
        <CaseEditor
          cases={casesArr}
          methods={methodsArr}
          organizations={orgsArr}
          thing={this.state.case}
          onSubmit={this.onSubmit.bind(this)}
        />
      );
    } else {
      return <div />;
    }
  }
}
