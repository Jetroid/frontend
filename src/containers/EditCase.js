import React, { Component } from 'react';
import styles from './Case.sass'
import CSSModules from 'react-css-modules'
import {injectIntl, intlShape} from 'react-intl'
import api from '../utils/api'
import { Field, reduxForm } from 'redux-form'
import ReactQuill from 'react-quill'
import QuillCSS from '../../node_modules/quill/dist/quill.snow.css'

import {
  AutoComplete,
  Checkbox,
  DatePicker,
  RadioButtonGroup,
  SelectField,
  Slider,
  TextField,
  Toggle
} from 'redux-form-material-ui'


const mapStateToProps = (state, ownProps) => {
  console.log("in mapStateToProps:", state, ownProps);
  return state
}
const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    onSubmit: (data) => {
      console.log('FORM PROPS', data, ownProps)
    }
  }
}
export const fields = [ 'title_en', 'body_en' ]

var toolbarOptions = [
  ['bold', 'italic', 'underline', 'strike'],        // toggled buttons
  ['blockquote', 'code-block'],

  [{ 'header': 1 }, { 'header': 2 }],               // custom button values
  [{ 'list': 'ordered'}, { 'list': 'bullet' }],
  [{ 'script': 'sub'}, { 'script': 'super' }],      // superscript/subscript
  [{ 'indent': '-1'}, { 'indent': '+1' }],          // outdent/indent
  [{ 'direction': 'rtl' }],                         // text direction

  [{ 'size': ['small', false, 'large', 'huge'] }],  // custom dropdown
  [{ 'header': [1, 2, 3, 4, 5, 6, false] }],

  [{ 'color': [] }, { 'background': [] }],          // dropdown with defaults from theme
  [{ 'font': [] }],
  [{ 'align': [] }],

  ['clean']                                         // remove formatting button
]

var _MyComponent = React.createClass({
  /* ... */

  _quillModules: {
    toolbar: '#toolbar'
    /* ... other modules */
  },

  _quillFormats: [ 
    'header',
    'bold', 'italic', 'underline', 'strike', 'blockquote',
    'list', 'bullet', 'indent',
    'link', 'image' 
  ],

  render: function() {
    return (
      <div className='_quill'>
        <div id="toolbar">
          <button className="ql-bold">Bold</button>
          <button className="ql-italic">Italic</button>
        </div>
        <ReactQuill theme='snow' 
                    modules={this._quillModules}
                    formats={this._quillFormats}
                    toolbar={false} // Let Quill manage toolbar
                    bounds={'._quill'}>
          <div key="editor"
                ref="editor"
                className="quill-contents border_solid_top"
                dangerouslySetInnerHTML={{__html:this.props.value}} />
        </ReactQuill>
      </div>
    )
  }
})

var MyComponent = CSSModules(_MyComponent, QuillCSS)

class _CaseForm extends Component {
  componentWillMount () {
    let component = this
    api.fetchCaseById(this.props.params.nodeID).then(function (json) {
      console.log('in EditCase.js componentWillMount, json = ', json[0])
      let data = json[0]
      component.setState({initialValues: data})
    })
  }

  render() {
    const { handleSubmit } = this.props
    console.log("props", this.props)
    return (
        <div>
          <div styleName='main-contents'>
            <div styleName='detailed-case-component'>
              <div styleName='sidebar'>
                <p styleName='case-location'>
                  country picker
                </p>
                <p styleName='sub-heading'>
                  Keywords
                </p>
                keyword picker
                <p styleName='sub-heading'>
                  Related Content
                </p>
                <div styleName='related-content'>
                  <a href='#'>Cases</a>
                  <a href='#'>Methods</a>
                  <a href='#'>Surveys</a>
                  <a href='#'>Datasets</a>
                </div>
              </div>
              <div styleName='main-area'>
                <div styleName='case-box'>
                  <div styleName='category'>
                    Case
                  </div>
                  <p styleName='case-title'>
                  case title
                  </p>
                  <form onSubmit={handleSubmit}>
                    <div>
                      <label htmlFor="title_en">Title</label>
                    </div>
                    <Field
                      component={TextField}
                      name="title_en" type="text"/>
                    <div>
                    <label htmlFor="body_en">Body</label>
                    </div>
                    <Field
                      component={MyComponent}
                      name="body_en" 
                      hintText="Remember the reader"
                      floatingLabelText="Case body"
                      multiLine={true}
                      rows={200}
                    />
                    <button type="submit">Submit</button>
                  </form>
                </div>
              </div>
          </div>
        </div>
      </div>
    )
  }
}


_CaseForm.propTypes = {
  intl: intlShape.isRequired
}

// Decorate the form component
let CaseForm = reduxForm({
  form: 'caseform', // a unique name for this form
}, mapStateToProps, mapDispatchToProps)(injectIntl(CSSModules(_CaseForm, styles)))

export default CaseForm

