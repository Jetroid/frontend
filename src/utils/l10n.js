import locales from "../../public/locales.json";

export function getBestMatchingMessages(locale) {
  if (locale in locales) {
    return locales[locale];
  }
  let [language] = locale.split("-");
  if (language in locales) {
    return locales[language];
  }
  return locales["en"];
}

export function getFirstBrowserLanguage() {
  let nav = window.navigator;
  let browserLanguagePropertyKeys = [
    "language",
    "browserLanguage",
    "systemLanguage",
    "userLanguage"
  ];
  let i = 0;
  let language = "";

  // support for HTML 5.1 "navigator.languages"
  if (Array.isArray(nav.languages)) {
    for (let i = 0; i < nav.languages.length; i++) {
      let language = nav.languages[i];
      if (language && language.length) {
        return language;
      }
    }
  }

  // support for other well known properties in browsers
  for (i = 0; i < browserLanguagePropertyKeys.length; i++) {
    language = nav[browserLanguagePropertyKeys[i]];
    if (language && language.length) {
      return language;
    }
  }

  return null;
}
