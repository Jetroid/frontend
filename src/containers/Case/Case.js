import React from 'react'
import './Case.css'
import {injectIntl, intlShape} from 'react-intl'
import {Link} from 'react-router'
import api from '../../utils/api'
import moment from 'moment'
import { Container, Row, Col } from 'reactstrap'
import CountryMap from '../../components/CountryMap'
import FloatingActionButton from 'material-ui/FloatingActionButton'
import ContentPencil from 'material-ui/svg-icons/image/edit'
import caseIconBookmark from '../../img/pp-case-icon-bookmark.png'
import caseIconSettings from '../../img/pp-case-icon-settings.png'
import caseIconFB from '../../img/pp-case-icon-fb.png'
import caseIconTW from '../../img/pp-case-icon-tw.png'
import caseIconShare from '../../img/pp-case-icon-share.png'


export class Case extends React.Component {
  componentWillMount () {
    let component = this
    api.fetchCaseById(this.props.params.nodeID).then(function (the_case) {
      component.setState({data: the_case, htmlBody: the_case.body})
    })
  }

  getInnerHTML () {
    let input = ''
    if (this.state && this.state.htmlBody) {
      input = this.state.htmlBody
    }
    return {__html: input}
  }

  render () {
    if (this.state && this.state.data) {
      let caseObject = this.state.data
      let tag, communication_mode = ''
      if (caseObject.specific_topic) {
        tag = <a href='#'>{caseObject.specific_topic}</a>
      }
      if (caseObject.communication_mode) {
        communication_mode = <a href='#'>{caseObject.communication_mode}</a>
      }
      let post_date = moment(caseObject.post_date).format('LL')
      let updated_date = moment(caseObject.updated_date).format('LL')
      let first_author = caseObject.authors[0]
      let locale = this.props.intl.locale
      let first_author_url = '/' + locale + '/users/' + first_author.id
      let first_author_name = first_author.name
      let last_author = caseObject.authors.slice(-1)[0]
      let last_author_name = last_author.name
      let last_author_url = '/' + locale + '/users/' + last_author.id
      let id = this.props.params.nodeID
      let editLink = `/${locale}/case/${id}/edit`
      // let editLink = (<Link to={`/${locale}/case/${id}/edit`} />)
      let awsUrl = process.env.REACT_APP_ASSETS_URL
      if (caseObject.lead_image) {
        var comma = caseObject.lead_image.search(',')
        var pic = awsUrl + encodeURIComponent(caseObject.lead_image.slice(9, comma-1))
      }
      if (caseObject.other_images) {
        var bracket = caseObject.other_images.search(']')
        var otherImg = awsUrl + encodeURIComponent(caseObject.other_images.slice(2, bracket-1))
      }

      return (
        <div>
          <div className='main-contents'>
            <Container className='detailed-case-component' fluid={true}>
              <Row>
                <Col md='3' className='hidden-sm-down sidepanel hidden-sm-down'>
                  <CountryMap city={caseObject.location.city} countrycode={caseObject.location.country} />
                  <p className='sub-heading'>
                    Keywords
                  </p>
                  <p className='sub-sub-heading'>
                    Tag:
                  </p>
                  <div className='tags'>
                  {tag}
                  </div>
                  <p className='sub-sub-heading'>
                    Specific Topic:
                  </p>
                  <div className='tags'>
                  {communication_mode}
                  </div>
                  <p className='sub-heading'>
                    Related Content
                  </p>
                  <div className='related-content'>
                    <a href='#'>Cases</a>
                    <a href='#'>Methods</a>
                    <a href='#'>Surveys</a>
                    <a href='#'>Datasets</a>
                  </div>
                </Col>
                <Col md='8' xs='12' className='main-area'>
                  <div className='case-box'>
                    <h2 className='category'>
                      Case
                    </h2>
                    <h2 className='case-title'>
                      {caseObject.title}
                    </h2>
                    { (pic && pic.length > awsUrl.length) ?
                      <div className='case-images'>
                        <img role='presentation' src={pic} />
                      </div>
                      :
                      (otherImg && otherImg.length > awsUrl.length) ?
                        <div className='case-images'>
                          <img role='presentation' src={otherImg} />
                        </div>
                      :
                      undefined
                    }
                    <div className='authorship-details'>
                      <p className='author-line'>
                        First submitted by&nbsp;
                        <a href={first_author_url}>
                          {first_author_name}
                        </a>
                      </p>
                      <p className='date-line'>
                      {post_date}
                      </p>
                      <p className='author-line'>
                        Most recent changes by&nbsp;
                        <a href={last_author_url}>
                          {last_author_name}
                        </a>
                      </p>
                      <p className='date-line'>
                      {updated_date}
                      </p>
                    </div>
                    <div className='case-html' dangerouslySetInnerHTML={{__html: caseObject.body}} />
                  </div>
                </Col>
                <Col md='1' className='case-tools hidden-sm-down'>
                  <div className='top-icons'>
                    <a href='#'><img src={caseIconBookmark} alt='' /></a>
                    <a href='#'><img src={caseIconSettings} alt='' /></a>
                    <a href='#'><img src={caseIconFB} alt='' /></a>
                    <a href='#'><img src={caseIconTW} alt='' /></a>
                    <a href='#'><img src={caseIconShare} alt='' /></a>
                  </div>
                </Col>
              </Row>
            </Container>
            <Link to={editLink}>
              <FloatingActionButton className='editButton'>
                <ContentPencil />
              </FloatingActionButton>
            </Link>
          </div>
        </div>)
    } else {
      return (<div></div>)
    }
  }
}

Case.propTypes = {
  intl: intlShape.isRequired
}

export default injectIntl(Case)
