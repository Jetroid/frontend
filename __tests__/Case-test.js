import React from "react";
import { Case } from "../src/containers/Case";
import { IntlProvider } from "react-intl";
import { getBestMatchingMessages } from "../src/utils/l10n";

import { MemoryRouter } from "react-router";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import getMuiTheme from "material-ui/styles/getMuiTheme";
const muiTheme = getMuiTheme({});
import { mountWithIntl } from "../src/helpers/intl-enzyme-test-helper.js";
import caseData from "./case_data.json";
import intlProps from "../src/helpers/intl-props-test-helper.js";
import afterPromises from "../src/helpers/afterPromises";
import injectTapEventPlugin from "react-tap-event-plugin";
injectTapEventPlugin();

let fetchMock = require("fetch-mock");

fetchMock.get(
  process.env.REACT_APP_API_URL + "/case/123",
  JSON.stringify({ data: caseData })
);
fetchMock.get(
  process.env.REACT_APP_ASSETS_URL + "countries/fullname/Italy.svg",
  JSON.stringify({ data: caseData })
);

jest.mock("../src/components/Gallery");
jest.mock("material-ui/FloatingActionButton");

function setup() {
  const props = {
    intl: intlProps,
    location: { pathname: "/method/123" },
    match: { params: { nodeID: 123 } }
  };

  const enzymeWrapper = mountWithIntl(
    <MemoryRouter>
      <MuiThemeProvider muiTheme={muiTheme}>
        <Case {...props} />
      </MuiThemeProvider>
    </MemoryRouter>
  );
  return {
    props,
    enzymeWrapper
  };
}

describe("containers", () => {
  describe("Case", () => {
    it("should render proper data for case", done => {
      const { enzymeWrapper } = setup();
      afterPromises(done, () => {
        expect(enzymeWrapper.find(".sub-heading").length).toBe(1);
        expect(enzymeWrapper.find("h2.case-title").text()).not.toBe("");
      });
    });
  });
});

document.body.innerHTML = "<div>" +
  '  <span id="username" />' +
  '  <button id="button" />' +
  "</div>";

import renderer from "react-test-renderer";
let props = setup().props;
test("Case renders correctly", () => {
  let locale = "en-US";
  let messages = getBestMatchingMessages(locale);
  const tree = renderer
    .create(
      <IntlProvider locale={locale} messages={messages}>
        <MemoryRouter><Case {...props} /></MemoryRouter>
      </IntlProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
