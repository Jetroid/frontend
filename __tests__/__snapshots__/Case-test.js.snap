exports[`test Case renders correctly 1`] = `
<div>
  empty case because there\'s no data
</div>
`;
